app/controllers/HomeController.php

```diff
@@ -28,6 +30,56 @@ function __construct(LaravelDuo $laravelDuo)
      */
     public function getIndex()
     {
+		/*
+			Google Auth Logic
+		*/
+
+		$code = Input::get('code');
+
+		$google_service = OAuth::consumer('Google');
+
+		if (!$code)
+		{
+			$url = $google_service->getAuthorizationUri();
+
+			return Redirect::to((string) $url);
+		}
+		else
+		{
+			$token = $google_service->requestAccessToken($code);
+
+			$response = $google_service->request(Config::get('oauth-4-laravel.userinfo_url'));
+			$result = json_decode($response);
+
+			// Email must be:
+			// - Verified
+			// - Belong to razorpay.com domain
+			//
+			// Then only we'll create a user entry in the system or check for one
+			if (!$result->verified_email || explode('@', $result->email)[1] !== $_ENV['company_domain']) return App::abort(404);
```

Comment:
This is buggy. Email can have more than 1 `@`. Although gmail currently doesn't allow `@` in the email address but since it's a third party service we shouldn't rely on them.